informer/informer.go

```diff
@@ -41,8 +41,8 @@ const (
 // Config represents the configuration used to create a new Informer.
 type Config struct {
 	// Dependencies.
-	BackOff    backoff.BackOff
-	RestClient rest.Interface
+	BackOff        backoff.BackOff
+	WatcherFactory WatcherFactory
```

Comment:
I eased the configuration and execution of the watch using a watcher factory. This was done in the first place when I tried to write useful unit tests. I couldn't find a way to do this and I ended up with the integration test introduced in this PR. I kept the interface/configuration improvements though.